### PR TITLE
Always use local variables when storing the options to prevent overri…

### LINKED ISSUE #753: [BUG] Adding an `organization` to the package config causes tests to fail silently.
### What happened?

Adding a `organization` to the `_viash.yaml` file causes tests to silently fail _when the test docker container does not exist yet._

### Steps to reproduce

**LOOK OUT! THIS EXAMPLE RUNS `docker system prune -a -f` and will remove all containers, volumes and caches from your system**
Let's setup a test case:
```
rm -r /tmp/test_bug
mkdir -p /tmp/test_bug
cd /tmp/test_bug
cat > _viash.yaml << 'HERE'
viash_version: 0.9.0-RC6

source: src
target: target
organization: "openpipelines-bio"
HERE

mkdir -p src/dummycomponent

cat > src/dummycomponent/config.vsh.yaml << 'HERE'

name: myscript
namespace: "dummycomponent"
test_resources:
  - type: python_script
    path: test_script.py
    text: |
      raise ValueError("TEST FAIL")

engines:
  - type: docker
    image: 'python:latest'

runners:
  - type: executable
HERE
```

And run this test component:

`docker system prune -a -f && viash test ./src/dummycomponent/config.vsh.yaml`

This gives:

```
====================================================================
+/tmp/viash_test_myscript_6536994418194033747/test_test_script/test_executable
[notice] Checking if Docker image is available at 'dummycomponent/myscript:test'
[warning] Could not pull from 'dummycomponent/myscript:test'. Docker image doesn't exist or is not accessible.
[notice] Building container 'dummycomponent/myscript:test' with Dockerfile
Traceback (most recent call last):
  File "/viash_automount/tmp/viash_test_myscript_6536994418194033747/test_test_script/tmp/viash-run-myscript-QmFUxD.py", line 31, in <module>
    raise ValueError("TEST FAIL")
ValueError: TEST FAIL
====================================================================
SUCCESS! All 1 out of 1 test scripts succeeded!
Cleaning up temporary directory
```

Running the test again, does produce the correct error:

```
viash test ./src/dummycomponent/config.vsh.yaml
```
```
====================================================================
+/tmp/viash_test_myscript_10126328146746982995/test_test_script/test_executable
Traceback (most recent call last):
  File "/viash_automount/tmp/viash_test_myscript_10126328146746982995/test_test_script/tmp/viash-run-myscript-4xWUSP.py", line 31, in <module>
    raise ValueError("TEST FAIL")
ValueError: TEST FAIL
====================================================================
ERROR! Only 0 out of 1 test scripts succeeded!
Unexpected error occurred! If you think this is a bug, please post
create an issue at https://github.com/viash-io/viash/issues containing
a reproducible example and the stack trace below.

viash - 0.9.0-RC6
Stacktrace:
java.lang.RuntimeException: Only 0 out of 1 test scripts succeeded!
	at io.viash.ViashTest$.apply(ViashTest.scala:137)
	at io.viash.Main$.mainCLI(Main.scala:256)
	at io.viash.Main$.mainCLIOrVersioned(Main.scala:124)
	at io.viash.Main$.main(Main.scala:59)
	at io.viash.Main.main(Main.scala)
```

But removing the container produces the erroneous behavior again:
```
docker system prune -a -f && viash test ./src/dummycomponent/config.vsh.yaml
```
```
====================================================================
+/tmp/viash_test_myscript_10345737742027411938/test_test_script/test_executable
[notice] Checking if Docker image is available at 'dummycomponent/myscript:test'
[warning] Could not pull from 'dummycomponent/myscript:test'. Docker image doesn't exist or is not accessible.
[notice] Building container 'dummycomponent/myscript:test' with Dockerfile
Traceback (most recent call last):
  File "/viash_automount/tmp/viash_test_myscript_10345737742027411938/test_test_script/tmp/viash-run-myscript-J7Ka4f.py", line 31, in <module>
    raise ValueError("TEST FAIL")
ValueError: TEST FAIL
====================================================================
SUCCESS! All 1 out of 1 test scripts succeeded!
Cleaning up temporary directory
```

### Expected behavior

I would expect viash to capture and return the correct exit code and not silently fail

### Relevant log output

_No response_

### Version

- OS: Ubuntu 24.04 LTS
- Java: openjdk 21.0.3 2024-04-16
- Viash 0.9.0-RC6

### Possible solution

_No response_

### Confirmation

- [X] I have searched the existing issues to make sure this is not a duplicate.
- [X] I have provided clear and concise information about the bug.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Grifs**
…ding in sub functions

## Describe your changes

Use `local` consistently when executing `save=$-; set +e` in the bash wrapper.
Essentially, the `-e` option wasn't being set again.

To give a more clear explanation of what's going on:
```
function foo2 {
	foo="foo2"
	echo "foo2: $foo"
}

function foo1 {
	local foo="foo1"
	foo2
	echo "foo1: $foo"
}

foo="foo"
foo1
echo "foo: $foo"
```

results in:
```
foo2: test2
foo1: test2
test: test
```

## Related issue(s)

Closes #753 

## Type of Change

- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] New functionality (non-breaking change which adds functionality)
- [ ] Major change (non-breaking change which modifies existing functionality)
- [ ] Minor change (non-breaking change which does not modify existing functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] This change requires a documentation update

## Checklist

Requirements:

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc.
- [x] I have performed a self-review of my code by checking the "Changed Files" tab.
- [x] My code follows the code style of this project.

Tests:

- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.

Documentation:

- [x] Proposed changes are described in the CHANGELOG.md.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by rcannood — CHANGES_REQUESTED** [review]
Could you

* Add a test
* Add `local` to all local variables in the bash functions

Not sure how I somehow missed this in the first place

**Comment by Grifs** [comment]
> Could you
> 
> * Add a test
> * Add `local` to all local variables in the bash functions
> 
> Not sure how I somehow missed this in the first place

yes, yes. working on it.

File: src/main/resources/io/viash/helpers/bashutils/ViashDockerFuns.sh [hunk 1]
```diff
@@ -13,9 +13,9 @@ function ViashDockerInstallationCheck {
   fi
 
   ViashDebug "Checking whether the Docker daemon is running"
-  save=$-; set +e
+  local save=$-; set +e
   docker_version=$(docker version --format '{{.Client.APIVersion}}' 2> /dev/null)
```
Comment (rcannood) [inline src/main/resources/io/viash/helpers/bashutils/ViashDockerFuns.sh:17]:
```suggestion
  local docker_version=$(docker version --format '{{.Client.APIVersion}}' 2> /dev/null)

**Review by rcannood — APPROVED** [review]
minor suggestion, otherwise LGTM!

File: src/main/resources/io/viash/helpers/bashutils/ViashDockerFuns.sh (hunk 1 above)
Comment (Grifs) [reply src/main/resources/io/viash/helpers/bashutils/ViashDockerFuns.sh:17]: ahh right! I spotted that before and then lost track of it. `docker_version` isn't used, so first I thought it must be used somewhere else, but it turns out it's not used at all

**Pull request merged**